mono/eglib/glib.h

```diff
@@ -54,6 +54,15 @@
 #define G_END_DECLS
 #endif
 
+// These macros are duplicated in mono-compiler.h, mono-publib.h, glib.h for maximum reach and consistency.
```

Comment:
Revert that, it has nothing to do here.

Reply:
Seems pretty straightforward to me: `MONO_BEGIN_DECLS` in public headers, `G_BEGIN_DECLS` everywhere else. We already do that for a few types. I am pretty optimistic that's not too complex.

Reply:
It is a rule, not terrible, can establish its own consistency, but when you suggested consistency, to me that meant something even simpler and more consistent — use just one of G_ or MONO_ w/o requiring to partition into public/private headers or be sensitive to such partitioning. I mean, you know, the PR already had a mix, which you objected to. I thought the idea was drive down/out the mix, not have a specific mix. To the casual observer this consistency will not be evident, it will just look like another inconsistency. But the importance of casual observers is unclear, granted.